Autogenerate attestation libraries using protoc

adityasaky opened the pull request:
This PR adds a GitHub Actions workflow to regenerate libraries using protoc. The action is triggered once a week as well as when something is pushed to main.

(We need the scheduled run as protoc is configured to use version `3.x` so we may have newer tooling even if there are no changes here for a week to main.)

marcelamelara commented on .github/workflows/make-protos.yml:
```diff
@@ -0,0 +1,39 @@
+  push:
+    branches:
+      - "main"
+  schedule:
```
Can you please add a comment describing the justification for needing the schedule?

TomHennen replied:
+1 to some more comments, even describing what it does a bit more would be nice.  E.g. It just sends the PR, right?  It doesn't auto-merge it?

adityasaky replied:
Added comments for the schedule and the create-pull-request step.

marcelamelara reviewed (commented):
Thanks! Do we need to update the documentation in docs/protos.md to remove the explicit requirement to regenerate the language bindings manually?

TomHennen commented on .github/workflows/make-protos.yml:
```diff
@@ -0,0 +1,39 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install protoc
+        uses: arduino/setup-protoc@ab6203da1c3118e4406048171b09238ad31ad73e
```
Do we really want the _arduino_ version of protoc?  Seems odd?

adityasaky replied:
It's not the arduino version, rather they're the creators of the setup-protoc action. It's certainly preferable to maintaining our own setup-protoc action. Presumably, they have similar codegen requirements and created the action for their needs?

TomHennen replied:
Got it.  Thanks for the explanation.

TomHennen commented on .github/workflows/make-protos.yml:
```diff
@@ -0,0 +1,39 @@
+        run: |
+          make protos
+      - name: Create PR
+        uses: peter-evans/create-pull-request@5b4a9f6a9e2af26e5f02351490b90d01eb8ec1e5
```
Is it secure to use this (random guy's?) thing for modifying code?  My guess is no one is going to provide much review on these PRs since they're machine generated.

adityasaky replied:
It's like the arduino thing above, it's a random guy but it's a pretty popular action for the job (https://github.com/peter-evans/create-pull-request/). Note that the action is only creating a commit and PR if the prior step `make protos` resulted in something to commit.

TomHennen reviewed (changes requested):
Thank you!

Also, my apologies if my other questions are daft, I've never done GitHub Actions before.

Finally, did we have some other documentation that suggests people regenerate the libs when modifying the protos?  I'd assume that would change now?

adityasaky commented:
Also amended the protos doc to instruct contributors not to regenerate libraries.

adityasaky commented:
Updated `go_setup` target in Makefile to be a `go install` of the Go generator. Also removed the explicit versioning so the one in go.mod is used by default.

marcelamelara reviewed (approved):
LGTM, thank you!
